### ISSUE TITLE
0.7 compile time doesn't work as well anymore

### ISSUE THREAD

**Issue by SimonDanisch** [JuliaLang/PackageCompiler.jl#110]
I'll soon push a branch with snoopcompile + sysimage working.
Meanwhile, the compilation speed improvements are not as drastic anymore.
I hope (am pretty sure?), this is just some flag we need to figure out.
```Julia
userimg = Pkg.dir("Makie", "test", "precompile.jl")
image_path = PackageCompiler.sysimg_folder()
PackageCompiler.build_sysimg(image_path, userimg, compilecache = "yes")
```
precompile.jl generated with `julia --trace-compilation=yes runtests.jl`.
Speed improvement is just from 80s to 20s, with a precompile file that should definitely have everything needed in it.

**Comment by SimonDanisch** [comment]
I see:
https://github.com/JuliaLang/julia/pull/27629

**Comment by SimonDanisch** [comment]
https://github.com/JuliaLang/julia/issues/27451

**Comment by SimonDanisch** [comment]
Good and bad news :D `build_sysimg` works exactly the same as PackageCompiler... So guess that means PackageCompiler is already up to date - but also means we don't get such large speed ups for some reason.

**Comment by NHDaly** [comment]
I talked some about my investigations into this here:
https://github.com/NHDaly/ApplicationBuilder.jl/issues/24

Perhaps I should merge that issue into this one, since this Repo sees a lot more traffic.

But, tl;dr, by setting both `--compile=all` when compiling, and `compile=min` in the jloptions in the C driver program at runtime, i was able to _actually_ avoid a bunch of recompilation at runtime. Without this, it was always recompiling everything for me.

Is this already something you're doing in your analysis?

**Comment by SimonDanisch** [comment]
*sigh*
```
signal (11): Segmentation fault
in expression starting at no file:0
```
But thanks, that gives me something to look into :)

**Comment by NHDaly** [comment]
Haha XD

yeah no problem. Weee the land of static compiling!

Also, if it's helpful, here's the changes i made to do what I talked about above:
https://github.com/NHDaly/ApplicationBuilder.jl/commit/ed97fc86a81981149c865989144d1983c446e39b

```julia
    PackageCompiler.build_executable(...
        compile="all",
    ...)
```
```C
    jl_options.compile_enabled = 3;  // 0 = "no"; 3 = "min"; 1 = "yes" (default)
```

**Comment by lucatrv** [comment]
Can this issue be closed?

**Comment by SimonDanisch** [comment]
@NHDaly you said you compiled your app successfully without JIT overhead in https://github.com/NHDaly/ApplicationBuilder.jl/issues/24#issuecomment-413299632... But I also can't see `compile="all"` on the master branch, and the branch you link never got merged - And for me `compile="all"` still segfaults.
Can you update us on the current status?

**Comment by KristofferC** [comment]
This is fixed (in Julia 1.3)
